src/java.desktop/unix/classes/sun/print/PrintServiceLookupProvider.java

```diff
@@ -278,23 +278,33 @@ public synchronized void refreshServices() {
               "Exception getting default printer : " + t);
         }
         if (CUPSPrinter.isCupsRunning()) {
-            try {
-                printerURIs = CUPSPrinter.getAllPrinters();
-                IPPPrintService.debug_println("CUPS URIs = " + printerURIs);
-                if (printerURIs != null) {
-                    for (int p = 0; p < printerURIs.length; p++) {
-                       IPPPrintService.debug_println("URI="+printerURIs[p]);
+            if(CUPSPrinter.useDomainSocketPathname()) {
+                printers = CUPSPrinter.getAllLocalPrinters();
+                printerURIs = new String[printers.length];
+                for (int p = 0; p < printers.length; p++) {
+                    printerURIs[p] = String.format("ipp://%s:%d/printers/%s",
+                            CUPSPrinter.getServer(), CUPSPrinter.getPort(), printers[p]);
+                    IPPPrintService.debug_println("URI="+printerURIs[p]);
+                }
```

Comment:
So it seems `getAllPrinters()` returns URIs (as line 291 suggests), and `getAllLocalPrinters()` returns printer names. You might want to move the above code into `getAllLocalPrinters()` to make it return URIs as well, just for consistency

Reply:
Updated. 
getCupsDefaultPrinters() method is moved to getAllPrinters().
Local printer names are converted to ipp uris.